### PR TITLE
Add array collapse function

### PR DESCRIPTION
**Pull request by omarkdev**
Adds array `collapse` function.

## PR Type
- [X] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [ ] Tools, Scripts & Automation (anything related to files in the scripts folder, Gatsby, website, Travis CI or Netlify)
- [ ] General, Typos, Misc. & Meta (everything related to content, typos, general stuff and meta files in the repository - e.g. the issue template)
- [ ] Other (please specifiy in the description above)

## Guidelines
- [X] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-starter/blob/master/CONTRIBUTING.md) document.

### CODE REVIEW

File: snippets/collapse.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+---
+
+Collapse a list of arrays into a single.
+
+Use `array_reduce()` to reduce each item into a single array, if the item
+is a array, merge the array with final item, if not, adds the item in a array and merge too.
+
+```php
+function collapse($array) 
+{
```
Comment (Chalarangelo) [inline snippets/collapse.md]: Please follow the snippet template's formatting for code. Indentation should be 2 extra spaces per indentation level.

File: snippets/collapse.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+		$itemToAdd = is_array($item) ? $item : [$item];
+
+		return array_merge($carry, $itemToAdd);
+	}, []);
+}
+```
+
+```php
+collapse([[1, 2, 3], [4, 5, 6], [7, 8, 9]]);
+// [1, 2, 3, 4, 5, 6, 7, 8, 9]
```
Comment (Chalarangelo) [inline snippets/collapse.md]: The output should be on the same line as the example.

File: snippets/collapse.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
+title:  collapse
+tags: array,beginner
+---
+
+Collapse a list of arrays into a single.
```
Comment (Chalarangelo) [inline snippets/collapse.md]: Rephrase:
```suggestion
Collapse a list of arrays into a single one.
```

File: snippets/collapse.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Use `array_reduce()` to reduce each item into a single array, if the item
+is a array, merge the array with final item, if not, adds the item in a array and merge too.
```
Comment (Chalarangelo) [inline snippets/collapse.md]: Do not hard break lines. Maybe rephrase to: 
```
Use `array_reduce()` to reduce each item into a single array, if the item is an array and merge with the result., otherwise, add the item in the result array.
```
(single line)

**Comment by gabbanaesteban** [comment]
What are the differences between [flatten](https://github.com/30-seconds/30-seconds-of-php#flatten) and this?

**Comment by Chalarangelo** [comment]
@gabbanaesteban None at all, as far as I can tell. Thanks for pointing this out, I'm closing on that basis (I thought we already had one, but I never checked).

**Comment by omarkdev** [comment]
@Chalarangelo @gabbanaesteban At first, both functions may look the same, but they are not! Following are examples:

```php
deepFlatten([[1, 2, 3, 'foo' => 'bar'], [4, 5, 6], [7, 8, 9]]);

Array
(
    [0] => 1
    [1] => 2
    [2] => 3
    [3] => bar
    [4] => 4
    [5] => 5
    [6] => 6
    [7] => 7
    [8] => 8
    [9] => 9
)
```

Already with the collapse:

```php
collapse([[1, 2, 3, 'foo' => 'bar'], [4, 5, 6], [7, 8, 9]]);

Array
(
    [0] => 1
    [1] => 2
    [2] => 3
    [foo] => bar
    [3] => 4
    [4] => 5
    [5] => 6
    [6] => 7
    [7] => 8
    [8] => 9
)
```

The `collapse` function keeps the keys, while the `flatten` function only takes the values.

**Comment by gabbanaesteban** [comment]
@omarkdev I see...
I don't have any use case for that but I get it.
